lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -885,10 +907,19 @@ class UncollapsedSlidingWindow2dPatternRewriter : public OpRewritePattern<T> {
         new_result_shape_array, op.getResult().getType().getElementType(),
         op.getResult().getType().getEncoding());
 
-    Operation *new_op = createMaxPool2dOp(
-        rewriter, mlir::cast<MaxPool2dOp>(op), input_reshape,
-        static_cast<int32_t>(input_shape[1]),
-        static_cast<int32_t>(input_shape[2]), new_result_type);
+    Operation *new_op = nullptr;
+
+    if (isa<MaxPool2dOp>(op)) {
+      new_op = createMaxPool2dOp(
+          rewriter, mlir::cast<MaxPool2dOp>(op), input_reshape,
+          static_cast<int32_t>(input_shape[1]),
+          static_cast<int32_t>(input_shape[2]), new_result_type);
+    } else if (isa<Conv2dOp>(op)) {
```

Comment:
Nit No need to check op type for else branch, only assert based on check above returning failure earlier.